src/containers/post/post.component.jsx

```diff
@@ -18,10 +20,12 @@ const Post = (props) => {
     }
   };
   useEffect(() => {
-    setCanUserViewVote(props.hasUserVoted);
+    if (props.hasUserVotedForA || props.hasUserVotedForB) {
+      setCanUserViewVote(true);
+    }
     setVoteACount(props.optionA.length);
     setVoteBCount(props.optionB.length);
-  }, [props.hasUserVoted]);
+  }, [props]);
```

Comment:
Are you sure you want to re-run thus `useEffect` for every single change in props?

Reply:
Ah good point